src/transaction/lock.rs

```diff
@@ -0,0 +1,106 @@
+use super::requests;
+use crate::pd::{PdClient, RegionVerId};
+use crate::request::KvRequest;
+use crate::{ErrorKind, Key, Result, Timestamp};
+
+use kvproto::kvrpcpb;
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+
+pub async fn resolve_locks(
```

Comment:
It looks to me like these functions should be methods on `PdClient`. Is there a reason they're not?

Reply:
Could you document what 'resolve' means

Reply:
> It looks to me like these functions should be methods on `PdClient`. Is there a reason they're not?

The requests are sent to TiKV instead of PD. So I think it should belong to TiKV requests.

Reply:
> Could you document what 'resolve' means

Ok.

Reply:
ISTM that this is a PD 'method' since we first query the timestamp, then `resolve_lock_with_retry` is a kv client method, although it is a bit weird that it has to figure out the region from pd (though I see why you'd do it this way). I think that there might be a nice way to factor these functions, at the moment it doesn't quite feel like the abstractions are right.